cmd/rhelemeter-server/main_e2e_test.go

```diff
@@ -159,6 +159,166 @@ func TestServerRHEL(t *testing.T) {
 	}
 }
 
+func TestServerRhelWithClientInfoFromHeaders(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	receiveServer := httptest.NewServer(mockedReceiver(t))
+	defer receiveServer.Close()
```

Comment:
Small suggestion here: might want to use `t.Cleanup(receiveServer.Close)`, as a `defer` won't run panic, but the first will.